x-pack/metricbeat/module/sql/query/dsn.go

```diff
@@ -5,17 +5,54 @@
 package query
 
 import (
+	"fmt"
 	"net/url"
 
 	"github.com/go-sql-driver/mysql"
+	"github.com/godror/godror"
+	"github.com/godror/godror/dsn"
 
 	"github.com/elastic/beats/v7/metricbeat/mb"
 )
 
+// ConnectionDetails contains all possible data that can be used to create a connection with
+// an Oracle db
+type ConnectionDetails struct {
+	Username string `config:"username"`
+	Password string `config:"password"`
+	Driver   string `config:"driver"`
+}
+
 // ParseDSN tries to parse the host
 func ParseDSN(mod mb.Module, host string) (mb.HostData, error) {
 	// TODO: Add support for `username` and `password` as module options
+	config := ConnectionDetails{}
+	if err := mod.UnpackConfig(&config); err != nil {
+		return mb.HostData{}, fmt.Errorf("error parsing config file: %w", err)
+	}
 
+	if config.Driver == "oracle" {
+		params, err := godror.ParseDSN(host)
+		if err != nil {
+			return mb.HostData{}, fmt.Errorf("error trying to parse connection string in field 'hosts': %w", err)
+		}
+
+		if params.Username == "" {
+			params.Username = config.Username
+		}
+
+		if params.Password.Secret() == "" {
+			params.StandaloneConnection = true
+			params.Password = dsn.NewPassword(config.Password)
+		}
+		return mb.HostData{
+			URI:          params.StringWithPassword(),
+			SanitizedURI: params.ConnectString,
+			Host:         params.String(),
+			User:         params.Username,
+			Password:     params.Password.Secret(),
+		}, nil
+	}
```

Comment:
I agree. 

The idea here is to start with one database (Oracle) and extend the adoption to other databases and certain other enhancements planned for 8.7.  

Following this, we can modify the santize() function to return dsn.ConnectionParams instead of string.

Reply:
The modification is expected to be taken up in 8.7 as part of MongoDB integration enhancements.